salt/utils/parsers.py

```diff
@@ -1044,12 +1037,13 @@ def start(self):
         self._install_signal_handlers()
 
     def _handle_signals(self, signum, sigframe):  # pylint: disable=unused-argument
-        msg = self.__class__.__name__
+        msg = [six.text_type(self.__class__.__name__)]
         if signum == signal.SIGINT:
-            msg += ' received a SIGINT.'
+            msg.append('received a SIGINT.')
```

Comment:
The string types do not have an `append` function.

Reply:
I missed the brackets around msg when it was being initialized. Still, allocating a list for this is slower:

```python
>>> timeit.Timer('''
... x = ['foo']
... x.append('bar')
... x = ' '.join(x)''').repeat()
[0.22678208351135254, 0.21661591529846191, 0.20710206031799316]
>>> timeit.Timer('''
... x = 'foo'
... x += " bar"''').repeat()
[0.08382701873779297, 0.054234981536865234, 0.05179405212402344]
```